middleware/common/include/common/network/byteorder.h

```diff
@@ -29,18 +30,12 @@ namespace casual
                template<typename type>
                struct transcode;
 
-               template<>
-               struct transcode<bool>
-               {
-                  static std::uint8_t encode( bool value) noexcept;
-                  static bool decode( std::uint8_t value) noexcept;
-               };
-
-               template<>
-               struct transcode<char>
+               // just passthrough 
+               template< typename T> requires concepts::any_of< T, bool, char, signed char, unsigned char>
+               struct transcode< T> 
```

Comment:
Should this (for completeness since c++20) include char8_t in the "any_of" list of types?

In C char8_t is a "typedef char8_t unsined char", but I think it is a distinct type in C++ 20.
 I don't think char8_t is used in casual so this comment is somewhat academical...

Reply:
I concur. And char8_t should be used :) But I think we do this when (if) we use std::u8string "everywhere". It's up for debate... @KristianIvarsson and I have talked a bit about it. We should take a stance which "encoding" we're handling/supporting (now we don't "care", and the user needs to have the same encoding on all machines). If we take a stance, then utf8 it is -> std::u8string "everywhere".